great_expectations/core/expectation_suite.py

```diff
@@ -249,9 +253,20 @@ def save(self) -> None:
         self._store.update(key=key, value=self)
 
     def is_added(self) -> ExpectationSuiteAddedDiagnostics:
-        return ExpectationSuiteAddedDiagnostics(
+        diagnostics = ExpectationSuiteAddedDiagnostics(
             errors=[] if self.id else [ExpectationSuiteNotAddedError(name=self.name)]
         )
+        if not diagnostics.is_added:
+            return diagnostics
+        return self._is_fresh()
+
+    def _is_fresh(self) -> ExpectationSuiteAddedDiagnostics:
```

Comment:
Don't we need to add the corresponding changes to `ValidationDefinition` and `Checkpoint`?

Reply:
This will happen in a follow up PR.